Timeout configuration ??

While indexing large document we get timeout is,

Caused by: org.apache.solr.client.solrj.impl.CloudSolrClient$RouteException: Timeout occurred while waiting response from server at: http://<>/_shard7_replica_n52
	at org.apache.solr.client.solrj.impl.CloudSolrClient.getRouteException(CloudSolrClient.java:125)
	at org.apache.solr.client.solrj.impl.CloudSolrClient.getRouteException(CloudSolrClient.java:46)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.directUpdate(BaseCloudSolrClient.java:551)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.sendRequest(BaseCloudSolrClient.java:1037)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.requestWithRetryOnStaleState(BaseCloudSolrClient.java:897)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.request(BaseCloudSolrClient.java:829)
	at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:211)
	at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:228)
	at com.github.jcustenborder.kafka.connect.solr.CloudSolrSinkTask.process(CloudSolrSinkTask.java:53)
	at com.github.jcustenborder.kafka.connect.solr.SolrSinkTask.put(SolrSinkTask.java:75)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:538)
	... 10 more
Caused by: org.apache.solr.client.solrj.SolrServerException: Timeout occurred while waiting response from server at: http://<>/shard7_replica_n52
	at org.apache.solr.client.solrj.impl.HttpSolrClient.executeMethod(HttpSolrClient.java:667)
	at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:262)
	at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:245)
	at org.apache.solr.client.solrj.impl.LBSolrClient.doRequest(LBSolrClient.java:368)
	at org.apache.solr.client.solrj.impl.LBSolrClient.request(LBSolrClient.java:296)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.lambda$directUpdate$0(BaseCloudSolrClient.java:525)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.solr.common.util.ExecutorUtil$MDCAwareThreadPoolExecutor.lambda$execute$0(ExecutorUtil.java:209)
	... 3 more

is it possible to introduce timeout parameter for Configuration like what we did for ZK connection timeout .

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 *.iml
 target
 .okhttpcache
+.idea

--- a/pom.xml
+++ b/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <groupId>com.github.jcustenborder.kafka.connect</groupId>
         <artifactId>kafka-connect-parent</artifactId>
-        <version>2.2.1-cp1</version>
+        <version>2.4.0</version>
     </parent>
     <artifactId>kafka-connect-solr</artifactId>
     <version>0.1-SNAPSHOT</version>

--- a/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkConnectorConfig.java
+++ b/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkConnectorConfig.java
@@ -28,6 +28,8 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
   public static final String ZOOKEEPER_CONNECT_TIMEOUT_CONFIG = "solr.zookeeper.connect.timeout.ms";
   public static final String ZOOKEEPER_CLIENT_TIMEOUT_CONFIG = "solr.zookeeper.client.timeout.ms";
   public static final String ZOOKEEPER_RETRY_EXPIRY_TIME_CONFIG = "solr.zookeeper.retry.expiry.time.ms";
+  public static final String SOLR_CONNECT_TIMEOUT_CONFIG = "solr.connect.timeout.ms";
+  public static final String SOLR_SOCKET_TIMEOUT_CONFIG = "solr.socket.timeout.ms";
 
 
   private static final String ZOOKEEPER_HOSTS_DOC = "Zookeeper hosts that are used to store solr configuration.";
@@ -36,12 +38,16 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
   private static final String ZOOKEEPER_CLIENT_TIMEOUT_DOC = "Set the timeout to the zookeeper ensemble in ms.";
   private static final String ZOOKEEPER_RETRY_EXPIRY_TIME_DOC = "This is the time to wait to refetch the " +
       "state after getting the same state version from ZK in ms.";
+  private static final String SOLR_CONNECT_TIMEOUT_DOC = "Set the connect timeout to the solr in ms.";
+  private static final String SOLR_SOCKET_TIMEOUT_DOC = "Set the solr read timeout on all sockets in ms.";
 
   public final List<String> zookeeperHosts;
   public final String zookeeperChroot;
   public final int zookeeperConnectTimeoutMs;
   public final int zookeeperClientTimeoutMs;
   public final int zookeeperRetryExpiryTimeMs;
+  public final int solrConnectTimeoutMs;
+  public final int solrSocketTimeoutMs;
 
   protected CloudSolrSinkConnectorConfig(Map<String, String> props) {
     super(config(), props);
@@ -50,6 +56,8 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
     this.zookeeperConnectTimeoutMs = getInt(ZOOKEEPER_CONNECT_TIMEOUT_CONFIG);
     this.zookeeperClientTimeoutMs = getInt(ZOOKEEPER_CLIENT_TIMEOUT_CONFIG);
     this.zookeeperRetryExpiryTimeMs = getInt(ZOOKEEPER_RETRY_EXPIRY_TIME_CONFIG);
+    this.solrConnectTimeoutMs = getInt(SOLR_CONNECT_TIMEOUT_CONFIG);
+    this.solrSocketTimeoutMs = getInt(SOLR_SOCKET_TIMEOUT_CONFIG);
   }
 
 
@@ -89,6 +97,20 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
                 .group(CONNECTION_GROUP)
                 .defaultValue(3000)
                 .build()
+        ).define(
+             ConfigKeyBuilder.of(SOLR_CONNECT_TIMEOUT_CONFIG, ConfigDef.Type.INT)
+                 .importance(ConfigDef.Importance.LOW)
+                 .documentation(SOLR_CONNECT_TIMEOUT_DOC)
+                 .group(CONNECTION_GROUP)
+                 .defaultValue(15000)
+                 .build()
+        ).define(
+             ConfigKeyBuilder.of(SOLR_SOCKET_TIMEOUT_CONFIG, ConfigDef.Type.INT)
+                 .importance(ConfigDef.Importance.LOW)
+                 .documentation(SOLR_SOCKET_TIMEOUT_DOC)
+                 .group(CONNECTION_GROUP)
+                 .defaultValue(120000)
+                 .build()
         );
   }
 }

--- a/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkTask.java
+++ b/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkTask.java
@@ -43,6 +43,8 @@ public class CloudSolrSinkTask extends SolrSinkTask<CloudSolrSinkConnectorConfig
     CloudSolrClient.Builder builder = new CloudSolrClient.Builder();
     builder.withZkHost(this.config.zookeeperHosts);
     builder.withZkChroot(this.config.zookeeperChroot);
+    builder.withConnectionTimeout(this.config.solrConnectTimeoutMs);
+    builder.withSocketTimeout(this.config.solrSocketTimeoutMs);
     this.client = builder.build();
     this.client.setZkConnectTimeout(this.config.zookeeperConnectTimeoutMs);
     this.client.setZkClientTimeout(this.config.zookeeperClientTimeoutMs);

